### PR TITLE
Hod clean removes localworkdir

### PR DESCRIPTION
**Pull request by ehiggs**
With this PR, hod clean will remove localworkdirs.

### CODE REVIEW

**Comment by boegel** [comment]
@ehiggs: please sync with master once #85 is merged, and update the section on `hod clean` to mention that it also cleans up the local work dirs

**Comment by ehiggs** [comment]
Jenkins: test this please.

**Comment by boegel** [comment]
@ehiggs: please sync with master to clean up a diff?

File: hod/cluster.py [hunk 1]
```diff
@@ -149,6 +149,11 @@ def cluster_jobid(label):
     return open(_cluster_info(label, 'jobid')).read()
 
 
+def cluster_workdir(label):
+    """Return workdir for cluster with specified label."""
+    return open(_cluster_info(label, 'workdir')).read()
```
Comment (boegel) [inline hod/cluster.py]: what if what you're trying to read doesn't exist?

be a little bit more careful?

```python
txt = None
filepath = _cluster_info(label, 'workdir')
if os.path.exists(filepath):
    try:
        txt = open(filepath).read()
    except IOError as err:
        raise SomeError("Unexpected error occured while trying to read %s: %s" % (filepath, err))
else:
    raise SomeError("some msg")

return txt
```

File: hod/cluster.py [hunk 2]
```diff
@@ -227,18 +232,26 @@ def mk_cluster_info(label, jobid):
...
             jobid_file.write(jobid)
     except IOError as err:
-        _log.error("Failed to write cluster info files: %s", err)
+        _log.error("Failed to write jobid file: %s", err)
+
+    try:
+        with open(os.path.join(info_dir, 'workdir'), 'w') as workdir_file:
+            workdir_file.write(workdir)
+    except IOError as err:
+        _log.error("Failed to write workdir file: %s", err)
```
Comment (boegel) [inline hod/cluster.py]: `log.error` doesn't exit or raise here, does it?

File: hod/cluster.py (hunk 1 above)
Comment (boegel) [inline hod/cluster.py]: oh, and use `with open`

File: hod/cluster.py [hunk 3]
```diff
@@ -227,18 +232,26 @@ def mk_cluster_info(label, jobid):
...
 
 
 def save_cluster_info(cluster_info):
-    """Save info (job ID, env script, ...) for this cluster in the cluster info dir."""
+    """
+    Save info (job ID, env script, ...) for this cluster in the cluster info dir.
+    cluster_info is a dict created by `gen_cluster_info` (not a ClusterInfo)
```
Comment (boegel) [inline hod/cluster.py]: `not a ClusterInfo instance` ?

File: hod/cluster.py [hunk 4]
```diff
@@ -271,6 +285,22 @@ def rm_cluster_info(label):
     print 'Removed cluster info directory %s for cluster labeled %s' % (info_dir, label)
 
 
+def rm_cluster_localworkdir(label):
+    """Remove a cluster localworkdir directory"""
+    jobid = cluster_jobid(label)
+    workdir = cluster_workdir(label)
+    jobid_workdir = os.path.join(workdir, 'hod', jobid)
+    try:
+        shutil.rmtree(jobid_workdir)
```
Comment (boegel) [inline hod/cluster.py]: only try to remove it if it exists? `os.path.exists(jobid_workdir)` ?

File: hod/cluster.py [hunk 5]
```diff
@@ -28,6 +28,7 @@
 @author: Kenneth Hoste (Universiteit Gent)
 """
 
+import errno
```
Comment (boegel) [inline hod/cluster.py]: not used, please drop

File: hod/cluster.py (hunk 1 above)
Comment (boegel) [inline hod/cluster.py]: discussed with @ehiggs off-pr, OK as is

This should never be used without `workdir` being there; if it is, it's a bug that will manifest itself as an IOError being raised, so fine.

File: hod/cluster.py (hunk 2 above)
Comment (boegel) [inline hod/cluster.py]: ping?

File: hod/cluster.py [hunk 6]
```diff
@@ -289,6 +316,8 @@ def post_job_submission(label, jobs):
     elif len(jobs) > 1:
         sys.stderr.write('Warning: More than one job found: %s\n' % str([j.jobid for j in jobs]))
     job = jobs[0]
-    print "Jobs submitted: %s" % str(job)
-    mk_cluster_info(label, job.jobid)
-
+    print "Job submitted: %s" % str(job)
+    try:
+        mk_cluster_info(label, job.jobid, workdir)
+    except OSError:
```
Comment (boegel) [inline hod/cluster.py]: IOError, mention the error, and exit > 0

```python
except (IOError, OSError) as err:
    sys.stderr.write("...: %s\n" % (label, err))
    sys.exit(1)
```

File: hod/local.py [hunk 7]
```diff
@@ -87,7 +87,11 @@ def main(args):
 
         _log.debug("Creating cluster info using label '%s'", label)
         cluster_info = gen_cluster_info(label, optparser.options)
-        save_cluster_info(cluster_info)
+        try:
+            save_cluster_info(cluster_info)
+        except OSError, e:
```
Comment (boegel) [inline hod/local.py]: here too, (also) catch IOError

**Pull request merged**